nc_test4/tst_xplatform2.c

```diff
@@ -476,16 +476,16 @@ main(int argc, char **argv)
        * type. */
       if (nc_def_compound(ncid, sizeof(struct s3), S3_TYPE_NAME, &s3_typeid)) ERR;
       if (nc_insert_array_compound(ncid, s3_typeid, VL_NAME,
-				   NC_COMPOUND_OFFSET(struct s3, data),
-				   vlen_typeid, 1, dimsizes)) ERR;
+   				   NC_COMPOUND_OFFSET(struct s3, data),
+   				   vlen_typeid, 1, dimsizes)) ERR;
 
 
       /* Write the output data as an attribute. */
       if (nc_put_att(ncid, NC_GLOBAL, S3_ATT_NAME, s3_typeid,
-		     DIM3_LEN, comp_array_of_vlen_of_comp_out)) ERR;
+   		     DIM3_LEN, comp_array_of_vlen_of_comp_out)) ERR;
 
-      /* How does it look? */
-      if (check_file_3(ncid, comp_array_of_vlen_of_comp_out)) ERR;
+      /* How does it look? Uncomment this line to see memory issue. */
+      /* if (check_file_3(ncid, comp_array_of_vlen_of_comp_out)) ERR; */
```

Comment:
If this line is uncommented, and the code built with address santizer, then a memory problem is revealed.